tests/qp/test_ensemble.py

```diff
@@ -199,6 +201,46 @@ def test_iterator(self):
             test_vals = ens_i.pdf(test_grid)
             assert np.allclose(check_vals, test_vals)
 
+    def test_quant_get_default_pdf_constructor_name(self):
+        """Test that the getter for pdf constructor name works"""
+        quantiles = np.linspace(0.001, 0.999, 16)
+        locations = np.linspace(0, 5, 16)
+        quant_dist = qp.quant(quants=quantiles, locs=locations)
+        self.assertEqual(quant_dist.dist.pdf_constructor_name, 'piecewise_linear')
+
+    def test_quant_get_default_pdf_constructor(self):
+        """Test that the getter for pdf constructor returns an AbstractQuantilePdfConstructor"""
+        quantiles = np.linspace(0.001, 0.999, 16)
+        locations = np.linspace(0, 5, 16)
+        quant_dist = qp.quant(quants=quantiles, locs=locations)
+        assert isinstance(quant_dist.dist.pdf_constructor, AbstractQuantilePdfConstructor)
+
+    def test_quant_change_pdf_constructor(self):
+        """Test that changing the pdf constructor works as expected"""
+        quantiles = np.linspace(0.001, 0.999, 16)
+        locations = np.linspace(0, 5, 16)
+        quant_dist = qp.quant(quants=quantiles, locs=locations)
+        quant_dist.dist.pdf_constructor_name = 'piecewise_constant'
+        self.assertEqual(quant_dist.dist.pdf_constructor_name, 'piecewise_constant')
+
+    def test_quant_change_pdf_constructor_raises(self):
+        """Verify that attempting to change the pdf constructor to one that
+        isn't in the dictionary, will raise an error."""
+        quantiles = np.linspace(0.001, 0.999, 16)
+        locations = np.linspace(0, 5, 16)
+        quant_dist = qp.quant(quants=quantiles, locs=locations)
+        with self.assertRaises(ValueError):
+            quant_dist.dist.pdf_constructor_name = 'drewtonian'
```

Comment:
This is a wonderful test string.

Reply:
Thanks, I can confidently say that a Drewtonian is not a thing now, and very likely never will be 😃